mysticeti-core/src/core_thread/simulated.rs

```diff
@@ -9,18 +9,18 @@ use crate::data::Data;
 use crate::syncer::{Syncer, SyncerSignals};
 use crate::types::{AuthoritySet, BlockReference};
 use crate::types::{RoundNumber, StatementBlock};
-use parking_lot::Mutex;
+use parking_lot::RwLock;
 
 pub struct CoreThreadDispatcher<H: BlockHandler, S: SyncerSignals, C: CommitObserver> {
-    syncer: Mutex<Syncer<H, S, C>>,
+    syncer: RwLock<Syncer<H, S, C>>,
```

Comment:
Yeah, I am not sure we need this change, since this is simulated code anyway and simulator runs in the single thread(so there is no difference between `Mutex` and `RwLock`)

Reply:
Yep this is not really needed, it has been transferred as part of some experimentation.